src/common/utils/resolve_reference.py

```diff
@@ -26,50 +51,15 @@ def _next_reference_part(reference: str) -> Tuple[str, Union[str, None], str]:
     return content, deliminator, remaining_reference
 
 
-def reference_to_reference_items(reference: str, items=None, prev_deliminator=None):
-    """Split up the reference into reference items."""
-
-    if len(reference) == 0:
-        return items
-
-    if not items:
-        items = []
-
-    content, deliminator, remaining_reference = _next_reference_part(reference)
-    if content == "":
-        # If the original reference starts with a deliminator (e.g. /)
-        # or two deliminators are next to each other (e.g. ([),
-        # then there is no content.
-        # The deliminator is extracted from the remaining reference.
-        # Continue resolve the remaining reference.
-        return reference_to_reference_items(remaining_reference, items, deliminator)
-
-    if "$" in content:  # By id
-        items.append(IdItem(content[1:]))
-    elif prev_deliminator == "/" and len(items) == 0:  # By root package
-        items.append(QueryItem(query=f"name={content},isRoot=True"))
-    elif prev_deliminator == "/":  # By package
-        items.append(AttributeItem("content"))
-        items.append(QueryItem(query=f"name={content}"))
-    elif prev_deliminator == "(" and deliminator == ")":  # By query
-        items.append(QueryItem(query=content))
-    elif prev_deliminator == "[" and deliminator == "]":  # By list index
-        items.append(AttributeItem(f"[{content}]"))
-    elif prev_deliminator == ".":  # By attribute
-        items.append(AttributeItem(content))
-    else:
-        raise Exception(f"Not supported reference format: {reference}")
-
-    return reference_to_reference_items(remaining_reference, items, deliminator)
-
-
 @dataclass
 class IdItem:
     """Points to an id."""
 
     id: str
 
-    def resolve(self, document: dict | list, data_source: DataSource, get_data_source: Callable) -> Tuple[dict, str]:
+    def resolve(
+        self, document: dict | list | None, data_source: DataSource, get_data_source: Callable
```

Comment:
document isn't used anywhere in the function

Reply:
Not my change. I believe this was done so all the "ReferenceItems" could have the same interfaces